scripts/preload/win32_wrap.rb

```diff
@@ -0,0 +1,369 @@
+		:OEM_2 => :UNKNOWN, :OEM_3 => :UNKNOWN, :OEM_4 => :UNKNOWN, :OEM_5 => :UNKNOWN,
+		:OEM_6 => :UNKNOWN, :OEM_7 => :UNKNOWN
+	}
+
+	WIN2SDL.default = :UNKNOWN
+end
+
+$win32KeyStates = nil
+
+module Graphics
+	class << self
+		alias_method(:win32wrap_update, :update)
+		def update
+			win32wrap_update
+			$win32KeyStates = nil
+		end
+	end
+end
+
+def get_raw_keystates
+	if $win32KeyStates == nil
+		$win32KeyStates = MKXP.raw_key_states
+	end
+
+	return $win32KeyStates
+end
+
+def common_keystate(vkey)
+	vkey_name = Scancodes::WIN32INV[vkey]
+
+	if vkey_name == :LBUTTON
+		return Input.press?(Input::MOUSELEFT) ? 1 : 0
+	elsif vkey_name == :RBUTTON
+		return Input.press?(Input::MOUSERIGHT) ? 1 : 0
+	elsif vkey_name == :MBUTTON
+		return Input.press?(Input::MOUSEMIDDLE) ? 1 : 0
+	end
+
+	offset = Scancodes::SDL[vkey_name]
+	return get_raw_keystates.getbyte(offset)
+end
+
+def memcpy_string(dst, src)
+	i = 0
+	src.each_byte do |b|
+		dst.setbyte(i, b)
+		i += 1
+	end
+end
+
+def state_pressed(states, sdl_scan)
+	return states.getbyte(Scancodes::SDL[sdl_scan]) == 1
+end
+
+def double_state(states, left, right)
+	return state_pressed(states, left) || state_pressed(states, right)
+end
+
+module Win32API_Impl
+	module User32
+		class Keybd_event
+			Seq = [
+				[0xA4, 0, 0, 0],
+				[0xD, 0, 0, 0],
+				[0xD, 0, 2, 0],
+				[0xA4, 0, 2, 0]]
+			Seq2 = [
+				[0x12, 0, 0, 0],
+				[0xD, 0, 0, 0],
+				[0xD, 0, 2, 0],
+				[0x12, 0, 2, 0]]
+			def initialize
+				@index = 0
+			end
+			def call(args)
+				seq = [args[0], args[1], args[2], args[3]]
+
+				if seq == Seq[@index] or seq == Seq2[@index]
+					@index += 1
+				else
+					@index = 0
+				end
+
+				if @index == 4
+					@index = 0
+					Graphics.fullscreen = !Graphics.fullscreen
+				end
+			end
+		end
+
+		class GetKeyState
+			def call(vkey)
+				return common_keystate(vkey[0])
+			end
+		end
+		class GetAsyncKeyState
+			PRESSED_BIT = (1 << 15)
+			def call(vkey)
+				return common_keystate(vkey[0]) == 1 ? PRESSED_BIT : 0
+			end
+		end
+		class GetKeyboardState
+			PRESSED_BIT = 0x80
+			def call(args)
+				out_states = args[0]
+				states = get_raw_keystates
+
+				Scancodes::WIN32.each do |name, val|
+					pressed = false
+
+					if name == :LBUTTON
+						pressed = Input.press?(Input::MOUSELEFT)
+					elsif name == :RBUTTON
+						pressed = Input.press?(Input::MOUSERIGHT)
+					elsif name == :MBUTTON
+						pressed = Input.press?(Input::MOUSEMIDDLE)
+					elsif name == :SHIFT
+						pressed = double_state(states, :LSHIFT, :RSHIFT)
+					elsif name == :MENU
+						pressed = double_state(states, :LALT, :RALT)
+					elsif name == :CONTROL
+						pressed = double_state(states, :LCONTROL, :RCONTROL)
+					else
+						scan = nil
+						if Scancodes::SDL.key?(name)
+							scan = name
+						else
+							scan = Scancodes::WIN2SDL[name]
+						end
+
+						pressed = state_pressed(states, name)
+					end
```

Comment:
```suggestion

				Scancodes::WIN32.each do |name, val|
					pressed = common_keystate(val) == 1
```
Removing redundant code.

Reply:
Just lemme know if you feel like it's ready and I'll add them in, since I'm in the mood to check on mkxp stuff again (finally)